macros/models/_sector/tokens/balances_daily.sql

```diff
@@ -0,0 +1,23 @@
+{%- macro balances_daily(balances_base) %}
+select
+    *,
+    {{ dbt_utils.generate_surrogate_key(['day', 'type', 'address', 'contract_address', 'token_id']) }} as unique_key
+from (
+    select
+        date_trunc('day', block_time) as day,
+        block_number,
+        block_time,
+        "type",
+        "address",
+        contract_address,
+        token_id,
+        amount,
+        row_number() OVER (partition by date_trunc('day', block_time), type, address, contract_address, token_id order by block_number desc) as row_number
+    from {{ balances_base }} balances
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('block_time')}}
+    {% endif %}
+) where row_number = 1
```

Comment:
is the row number to be safe, or known duplicate issues?

Reply:
This is to pick the latest value for each day, it's a way to do the aggregation it self instead of using `max/max_by`